packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -294,8 +295,27 @@ export default class SalesforceCRMService implements CRM {
     });
   };
 
-  private salesforceDeleteEvent = async (uid: string) => {
+  private salesforceDeleteEvent = async (uid: string, event: CalendarEvent) => {
+    const appOptions = this.getAppOptions();
     const conn = await this.conn;
+
+    if (appOptions?.onCancelWriteToEventRecord) {
+      const fieldsToWriteTo = appOptions?.onCancelWriteToEventRecordFields;
+
+      // If the option is enabled then don't delete the event record
+      if (!fieldsToWriteTo || !Object.keys(fieldsToWriteTo)) {
+        return Promise.resolve();
+      }
```

Comment:
If the option is enabled, we don't need to make any calls to Salesforce to write or delete the event record